pkg/authentication/firebase.go

```diff
@@ -48,3 +49,35 @@ func NewFirebaseWithTokenVerifier(firebaseAuth FirebaseTokenVerifier) Authentica
 		firebaseAuth: firebaseAuth,
 	}
 }
+
+// firebaseAuth uses the firebase application to refresh the keys used to verify the token signature.
+// The firebase.App has an internal mechanism to avoid repeating this operation for every request.
+type firebaseAuth struct {
+	client *auth.Client
+}
+
+// VerifyIDToken gets a new public key in case a key rotation has been requested, and verifies the given token.
+func (auth *firebaseAuth) VerifyIDToken(ctx context.Context, idToken string) (*auth.Token, error) {
+	return auth.client.VerifyIDToken(ctx, idToken)
+}
+
+// NewFirebase initializes a new FirebaseTokenVerifier implementation using a Firebase application, and it's in
+// charge of refreshing the public key used to verify tokens every time a new key rotation happens.
+//
+//	fbAuth, err := NewFirebase(app)
```

Comment:
What these code comments for? Should they be removed?

Reply:
Thanks for the question. They serve as an example on how to use the library, they look like this in Goland:

![image](https://github.com/gazebo-web/auth/assets/5170276/ef2b8e83-9859-44c0-98d1-3a18451ba9aa)